doc/md/writing-motoko/pattern-matching.md

```diff
@@ -100,6 +100,22 @@ Some types contain just a single value. We call these singleton types. Examples
 ### Exhaustiveness (coverage) checking
 
-
 At runtime, a switch expression may wind up scrutinizing a value to which none of its alternative patterns apply, generating an undesired trap.
-To detect the possibility of such runtime failures, the Motoko compiler checks for the exhaustiveness of pattern matching by keeping track of the covered shape of the scrutinee. The compiler issues a warning for any non-covered scrutinees. Motoko even constructs a helpful example of a scrutinee that is not matched. A useful by-product of the exhaustiveness check is that it identifies and warns about dead or redundant alternatives that can never be matched.
\ No newline at end of file
+To detect the possibility of such runtime failures, the Motoko compiler checks for the exhaustiveness of pattern matching by keeping track of the covered shape of the scrutinee. The compiler issues a warning for any non-covered scrutinees. Motoko even constructs a helpful example of a scrutinee that is not matched. A useful by-product of the exhaustiveness check is that it identifies and warns about dead or redundant alternatives that can never be matched.
+
+## Refutable patterns and dealing with non-matching data
+
+Numerous times one is only interested in specially formed data and desires to summarily handle all non-matching forms. The `let`-`else` construct is designed precisely for this pupose. Whereas the regular destructuring `let` allows to focus on a single given pattern, it invariably traps if the right hand side data doesn't match it (this is also warned at compile-time). The `else` clause gives the programmer a way to deal with refuted matches, such as bailing out of the processing or logging a message before trapping. As such, `let`-`else` is similar to a two-`case` `switch` in a compact form, that additionally doesn't force the indentation of the processing logic following it.
+
+The below example illustrates how you can write a non-indenting `if`-`else` by resorting to a `let`-`else` in your code:
+
+``` motoko
+let true = isLoggedIn(customer) else return;
+// process message for logged-in customer
+```
+
+The expression (or block) following the `else` must be of type `None`, thus signifying that its execution trajectory mustn't contribute to the code immediately following the `let` declaration.
+
+## Option blocks for streamlined processing of optional data
+
+Pattern matching on optional data (of type `?T`) is a preferred technique of avoiding the dreaded `null`-exeption problems known from other languages. However `switch`ing on several options can lead to tedious coding and deeply nested sources. To remedy these problems, Motoko provides *option blocks* (`do ? { ... }`) that allow safe unwrapping of options using a postfix `!` operator. Every use of the `!` in the block corresponds to a `switch` on some option, with the additional short-cutting behaviour that if `!` is applied to a `null` value, the entire block stops evaluation and immediately returns `null`.
```

Comment:
```suggestion
The expression (or block) following the `else` must be of type `None` signifying that its execution trajectory mustn't contribute to the code immediately following the `let` declaration.

## Option blocks for streamlined processing of optional data

Pattern matching on optional data (of type `?T`) is a preferred technique for avoiding the dreaded `null`-exeption problems known from other languages. However,' switch'ing  on several options can lead to tedious coding and deeply nested sources. To remedy these problems, Motoko provides *option blocks* (`do ? { ... }`) that allow safe unwrapping of options using a postfix `!` operator. Every use of the `!` in the block corresponds to a `switch` on some option, with the additional short-cut behavior that if `!` is applied to a `null` value, the entire block stops evaluation and immediately returns `null`.
```

Reply:
Maybe the word I was searching for is _short-circuiting_. @crusso?